sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/MessageUtils.java

```diff
@@ -253,4 +275,75 @@ private static TransactionalState getTransactionState(ByteBuffer transactionId,
         transactionalState.setOutcome(outcome);
         return transactionalState;
     }
+
+    public static ServiceBusMessage traceMessageSpan(ServiceBusMessage serviceBusMessage,
+        Context messageContext, String hostname, String entityPath, TracerProvider tracerProvider) {
+        Optional<Object> eventContextData = messageContext.getData(SPAN_CONTEXT_KEY);
+        if (eventContextData.isPresent()) {
+            // if message has context (in case of retries), don't start a message span or add a new context
+            return serviceBusMessage;
+        } else {
+            // Starting the span makes the sampling decision (nothing is logged at this time)
+            Context newMessageContext = messageContext
+                .addData(AZ_TRACING_NAMESPACE_KEY, AZ_TRACING_NAMESPACE_VALUE)
+                .addData(ENTITY_PATH_KEY, entityPath)
+                .addData(HOST_NAME_KEY, hostname);
```

Comment:
Should this also include the queue name, topic name etc?

Reply:
So EntityPath can be either queue or topic.